src/Network/Network/AzureFirewall/NetworkRule/NewAzureFirewallNetworkRuleCommand.cs

```diff
@@ -39,16 +41,60 @@ public class NewAzureFirewallNetworkRuleCommand : NetworkBaseCmdlet
 
         [Parameter(
             Mandatory = true,
+            ParameterSetName = AzureFirewallApplicationRuleParameterSets.Default,
             HelpMessage = "The source addresses of the rule")]
-        [ValidateNotNullOrEmpty]
+        [Parameter(
+            Mandatory = false,
+            ParameterSetName = AzureFirewallApplicationRuleParameterSets.ByDestinationAddress,
+            HelpMessage = "The destination addresses of the rule")]
+        [Parameter(
+            Mandatory = false,
+            ParameterSetName = AzureFirewallApplicationRuleParameterSets.ByDestinationIpGroup,
+            HelpMessage = "The destination ipgroup of the rule")]
```

Comment:
The help message is miss match with the parameter.

Reply:
Which line?

Reply:
Re-ordered the parameter set so that the help message matches visually